packages/styled-components/src/constants.js

```diff
@@ -13,11 +13,25 @@ export const SC_VERSION = __VERSION__;
 
 export const IS_BROWSER = typeof window !== 'undefined' && 'HTMLElement' in window;
 
-export const DISABLE_SPEEDY =
-  (typeof SC_DISABLE_SPEEDY === 'boolean' && SC_DISABLE_SPEEDY) ||
-  (typeof process !== 'undefined' &&
-    (process.env.REACT_APP_SC_DISABLE_SPEEDY || process.env.SC_DISABLE_SPEEDY)) ||
```

Comment:
Let's please keep this to a simpler change of `=== "true"` or another hardcoded string.

`typeof process.env.REACT_APP_SC_DISABLE_SPEEDY !== 'undefined'` is redundant,
`process.env.REACT_APP_SC_DISABLE_SPEEDY.toLowerCase()` defeats static optimisations (Webpack / Terser / etc won't be able to statically compile this away anymore)

Cheers :)

Reply:
@kitten better?